dpnp/dpnp_algo/dpnp_algo_indexing.pyx

```diff
@@ -184,20 +185,38 @@ cpdef dpnp_place(dparray arr, dparray mask, vals):
             counter += 1
 
 
-cpdef dpnp_put(input, ind, v):
+cpdef dpnp_put(dparray input, ind, v):
```

Comment:
``` suggestion
cpdef dpnp_put(dparray input_, ind, v):
```
In python/cython `input` is not a keyword. Instead, it is a built-in function. You can create a variable with the name `input`. But doing it is a bad practice because it overshadows the built-in (makes it unusable in the current scope). Use suggested convention, if you must use the name `input`.